tools/IoTEdgeDevOps/DevOpsLib/BuildManagement.cs

```
@@ -38,25 +39,62 @@ public async Task<IList<VstsBuild>> GetLatestBuildsAsync(HashSet<BuildDefinition
 
-            if (!result.ContainsKey("count") || (int) result["count"] <= 0)
+            if (!result.ContainsKey("count") || (int)result["count"] <= 0)
             {
                 return buildDefinitionIds.Select(i => VstsBuild.CreateBuildWithNoResult(i, branchName)).ToList();
             }
 
             Dictionary<BuildDefinitionId, VstsBuild> latestBuilds = JsonConvert.DeserializeObject<VstsBuild[]>(result["value"].ToString()).ToDictionary(b => b.DefinitionId, b => b);
             return buildDefinitionIds.Select(i => latestBuilds.ContainsKey(i) ? latestBuilds[i] : VstsBuild.CreateBuildWithNoResult(i, branchName)).ToList();
         }
+
+        public async Task<IList<VstsBuild>> GetBuildsAsync(HashSet<BuildDefinitionId> buildDefinitionIds, string branchName, DateTime minTime, int? maxBuildsPerDefinition = null)
+        {
+            ValidationUtil.ThrowIfNulOrEmptySet(buildDefinitionIds, nameof(buildDefinitionIds));
+            ValidationUtil.ThrowIfNullOrWhiteSpace(branchName, nameof(branchName));
+
+            // TODO: need to think about how to handle unexpected exception during REST API call
+            string requestPath = string.Format(LatestBuildPathSegmentFormat, this.accessSetting.Organization, this.accessSetting.Project);
+            IFlurlRequest latestBuildRequest = GetBuildsRequestUri(buildDefinitionIds, branchName, requestPath, minTime, maxBuildsPerDefinition)
+                .WithBasicAuth(string.Empty, this.accessSetting.PersonalAccessToken);
+
+            string resultJson = await latestBuildRequest.GetStringAsync().ConfigureAwait(false);
+            JObject result = JObject.Parse(resultJson);
+
+            if (!result.ContainsKey("count") || (int)result["count"] <= 0)
+            {
+                return buildDefinitionIds.Select(i => VstsBuild.CreateBuildWithNoResult(i, branchName)).ToList();
+            }
+
+            return JsonConvert.DeserializeObject<VstsBuild[]>(result["value"].ToString()).ToList();
```

Do we need to do `result.ContainsKey("value")` check before accessing it?

---

let it fail if it doesn't exist; means something wrong with VSTS API result.